test/controllers/LongerCommodityCodeControllerSpec.scala

```diff
@@ -69,271 +69,292 @@ class LongerCommodityCodeControllerSpec extends SpecBase with MockitoSugar {
+            .success
+            .value
+          val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
 
-      val application = applicationBuilder(userAnswers = None).build()
+          running(application) {
+            val request =
+              FakeRequest(POST, longerCommodityCodeRoute)
+                .withFormUrlEncodedBody(("value", ""))
 
-      running(application) {
-        val request = FakeRequest(GET, longerCommodityCodeRoute)
+            val boundForm = form.bind(Map("value" -> ""))
 
-        val result = route(application, request).value
+            val view = application.injector.instanceOf[LongerCommodityCodeView]
+
+            val result = route(application, request).value
+
+            status(result) mustEqual BAD_REQUEST
+            contentAsString(result) mustEqual view(boundForm, NormalMode, shortCommodity, testRecordId)(
+              request,
+              messages(application)
+            ).toString
+          }
+        }
+
+        "when correct data format but wrong data is submitted" in {
+          val mockGoodsRecordConnector = mock[GoodsRecordConnector]
+          when(mockGoodsRecordConnector.getRecord(any(), any())(any()))
+            .thenReturn(Future.successful(goodsRecordResponse(Instant.now, Instant.now)))
+
+          val mockOttConnector = mock[OttConnector]
+          val userAnswers      = emptyUserAnswers
+            .set(RecordCategorisationsQuery, recordCategorisationsShortCommodity)
+            .success
+            .value
+            .set(LongerCommodityCodePage(testRecordId), "answer")
+            .success
+            .value
+
+          when(
+            mockOttConnector.getCommodityCode(anyString(), any(), any(), any(), any(), any())(any())
+          ) thenReturn Future
+            .failed(
+              UpstreamErrorResponse(" ", NOT_FOUND)
+            )
+
+          val application = applicationBuilder(userAnswers = Some(userAnswers))
+            .overrides(
+              bind[OttConnector].toInstance(mockOttConnector),
+              bind[GoodsRecordConnector].toInstance(mockGoodsRecordConnector)
+            )
+            .build()
+
+          running(application) {
+            val request =
+              FakeRequest(POST, longerCommodityCodeRoute)
+                .withFormUrlEncodedBody(("value", "1234"))
+
+            val boundForm = form.copy(errors = Seq(elems = FormError("value", "longerCommodityCode.error.invalid")))
+
+            val view = application.injector.instanceOf[LongerCommodityCodeView]
+
+            val result = route(application, request).value
+
+            status(result) mustEqual BAD_REQUEST
+            contentAsString(result) mustEqual view(boundForm, NormalMode, shortCommodity, testRecordId)(
+              request,
+              messages(application)
+            ).toString
+
+            verify(mockOttConnector, times(1)).getCommodityCode(any(), any(), any(), any(), any(), any())(any())
```

Comment:
may be I think we should start removing times(1) from newly added tests as its already defaulted times(1) so we don't need to explicitly add it here. :)

Reply:
Fair 🙂